src/components/scenes/TransactionDetailsScene.js

```diff
@@ -175,19 +175,25 @@ export class TransactionDetailsComponent extends React.Component<Props, State> {
 
   onChangeFiat = (amountFiat: string) => this.setState({ amountFiat })
   openFiatInput = () => {
+    const {
+      guiWallet: { fiatCurrencyCode }
+    } = this.props
     Airship.show(bridge => (
-      <TransactionDetailsFiatInput
+      <TextInputModal
         bridge={bridge}
-        currency={this.props.guiWallet.fiatCurrencyCode}
-        amount={this.state.amountFiat}
-        onChange={this.onChangeFiat}
+        initialValue={this.state.amountFiat}
+        inputLabel={sprintf(s.strings.transaction_details_fiat_modal_header, fiatCurrencyCode)}
+        returnKeyType="send"
+        keyboardType="decimal-pad"
+        title={s.strings.transaction_details_fiat_label}
+        onSubmit={async amountInput => {
+          const amountCleaned = amountInput.replace(',', '.').replace(/[^\d.,]/g, '')
+          const amountFiat = bns.toFixed(amountCleaned, 2, 2)
+          await this.onChangeFiat(amountFiat)
+          return true
+        }}
```

Comment:
- This needs to be moved to utils.js 
- Create unit tests for cleaning amountInput
- Format should follow $NUMBER1.$NUMBER2, everything after $NUMBER2 gets removed
- If $NUMBER1 is a number and $NUMBER2 is not a number, then ignore $NUMBER2
- If $NUMBER1 is not a number return 0.00
- If return value evaluates to 0 (null / undefined / invalid), then return 0.00